R/fread.R

```diff
@@ -319,20 +320,9 @@ yaml=FALSE, autostart=NA)
     for (j in cols_to_factor) set(ans, j=j, value=as_factor(.subset2(ans, j)))
   }
 
-  if (!is.null(select)) {
-    if (is.numeric(select)) {
-      if (length(o <- forderv(select))) {
-        rank = integer(length(o))
-        rank[o] = 1:length(o)
-        setcolorder(ans, rank)
-      }
-    } else {
-      if (!identical(names(ans), select)) {
-        reorder = select[select %chin% names(ans)] # any missing columns are warned about in freadR.c and skipped
-        setcolorder(ans, reorder)
-      }
-    }
-  }
+  selectOrder = attr(ans, "selectOrder")  # only present if select= was specified
```

Comment:
this can't be done at the C level directly? rather than passing from C/fread to R back to C/setcolorder?

`setcolorder` ensures:

 - `!anyDuplicated(neworder)` [should be true of `selectOrder`]
 - `length(neworder) == length(x)` [ditto]
- `if (is.character(neworder))` check duplicates [`selectOrder` is `INTSXP` so this branch irrelevant]

So it seems OK to do from C?

Reply:
Good idea. Yes would be less moving parts done at C level. Ok will do